openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -711,6 +711,26 @@ def password_reset_logistration(request, **kwargs):
             form.save()
             reset_status = True
 
+            if 'is_account_recovery' in request.GET:
+                try:
+                    old_primary_email = user.email
+                    user.email = user.account_recovery.secondary_email
```

Comment:
Don't you think we should check secondary email is set or not. E.g learner has't set the account recovery email but he tries to recover the account.

Reply:
Going through the code it seems forget password code generates reset link based on 
1) When user has a primary but no secondary and user inserts secondary. In this case no password link is sent. System tries to treat the email as primary and tries to fetch the user but throws ObjectDoesNotExist error, as there is no user having a secondary email address.
2) When user has a primary and connected secondary and user inserts secondary. Link is sent to secondary with in_account_recovery queryparam.
3) When user has a primary and no connected secondary and inserts primary. Link is sent to primary without in_account_reovery queryparam.

So what happens when we have the link based on above info;
1) If the user has no associated secondary record, he/she will never get **is_account_recovery** queryparam so it will never hit this part of the code.
2) If somehow users hacks the url and send the queryparam by him/her self, then we have a try/catch case and only email is not send, but other functionality of reset password will be successful.

In future i think we have to see how to fix pt#1 of the system